circuit_knitting/cutting/cutting_reconstruction.py

```diff
@@ -79,22 +100,33 @@ def reconstruct_expectation_values(
         label: ObservableCollection(subobservables)
         for label, subobservables in subobservables_by_subsystem.items()
     }
+    sorted_subsystems = sorted(subsystem_observables.keys())
 
-    # Assign each weight's sign and calculate the expectation values for each observable
-    for i, coeff in enumerate(coefficients):
-        sorted_subsystems = sorted(subsystem_observables.keys())  # type: ignore
+    # Reconstruct the expectation values
+    for i in range(len(coefficients)):
```

Comment:
```suggestion
    for i, coeff in enumerate(coefficients):
```
will enable a line to be dropped ...